src/Units/Unit.java

```diff
@@ -46,27 +47,35 @@ public Unit(Commander co, UnitModel um)
 		if (model.holdingCapacity > 0)
 			heldUnits = new Vector<Unit>(model.holdingCapacity);
 	}
-	
-	public void initTurn()
-	{
+
+	public void initTurn(Location locus) {
 		isTurnOver = false;
 		fuel -= model.idleFuelBurn;
-		if (captureTarget != null && captureTarget.getResident() != this)
-		{
+		if (captureTarget != null && captureTarget.getResident() != this) {
 			captureTarget = null;
 			captureProgress = 0;
 		}
+		if (HP < 9.99) {
+			if (model.canRepairOn(locus) && locus.getOwner() == CO) {
+				int cost = model.moneyCost * Math.min(10 - getHP(), 2)/10; // 1/10th the cost to repair fully
```

Comment:
I don't like magic numbers. I assume the 10 from (10 - getHP()) is max health? Use the variable instead, so we know why. Also I'm not sure your comment makes sense; healing 2 HP is 1/10 the cost of healing 10HP? It looks like you are doing proportional healing, i.e. cost for 2 HP is ((unit cost / maxHP) * 2), which doesn't match that comment.